os/drivers/video/video_null_driver.c

```diff
@@ -237,6 +262,87 @@ struct v4l2_mapping_s ctrl_mapping[] = {
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+
+/****************************************************************************
+ * Name: video_null_set_worker_delay
+ *
+ * Description:
+ *   This function calculates the delay requred for the worker thread
+ *   to invoke a callback based on the resolution chosen and the
+ *   video frames-per-second required.
+ *
+ * Input Paramaters:
+ *   priv - pointer to null_priv_t object holding the frame
+ *   resolution attributes.
+ *
+ * Returned Values:
+ *   OK on success, appropriate error code otherwise.
+ ****************************************************************************/
+static int video_null_set_worker_delay(FAR null_priv_t *priv)
+{
+	int i;
+	struct v4l2_format_s *format;
+	if (priv == NULL) {
+		videodbg("Unable to set worker delay duration!!!\n");
+	}
+
+	format = priv->cur_format;
+
+	/* Delay depends on frame size to be read from file to the transfer buffer */
+	/* Getting the time delay for the default 30fps transfer based on resolution */
+	if (format == NULL) {
+		videodbg("No current format set, using default format instead\n");
+		format = priv->def_format;
+	}
+
+	if (!priv->cur_frame) {
+		priv->w_delay = W_DELAY_DEFAULT;
+		videodbg("No current frame set, continue with default worker_delay\n");
+		return OK;
+	}
+
+	for (i = 0; i < format->framecount; i++) {
+		if (format->frames[i].wHeight == priv->cur_frame->wHeight) {
+			break;
+		}
+	}
+
+	if (i == format->framecount) {
+		videodbg("No matching standard frame resolution found, kindly check");
+		return -EINVAL;
+	}
+
+	priv->w_delay = std_w_delays[i];
+	videodbg("Frame Interval = %d/%d\n", priv->cur_fintvl.numerator, priv->cur_fintvl.denominator);
+	/* Adding the excess delay required for transfer at less than 30fps */
+
+	switch (priv->cur_fintvl.numerator) {
```

Comment:
3. Delay is increased by a certain amount if transfer is required slower than 30fps.